crates/transport/src/boxed.rs

```diff
@@ -1,32 +1,57 @@
 use crate::{Transport, TransportError, TransportFut};
 use alloy_json_rpc::{RequestPacket, ResponsePacket};
-use std::fmt;
+use std::{any::TypeId, fmt};
 use tower::Service;
 
+#[allow(unnameable_types)]
+mod private {
+    pub trait Sealed {}
+    impl<T: super::Transport + Clone> Sealed for T {}
+}
+
+/// Trait for converting a transport into a boxed transport.
+///
+/// This trait is sealed and implemented for all types that implement
+/// [`Transport`] + [`Clone`].
+pub trait IntoBoxTransport: Transport + Clone + private::Sealed {
+    /// Boxes the transport.
+    fn into_box_transport(self) -> BoxTransport;
+}
+
+impl<T: Transport + Clone> IntoBoxTransport for T {
+    fn into_box_transport(self) -> BoxTransport {
+        // "specialization" to re-use `BoxTransport`.
+        if TypeId::of::<T>() == TypeId::of::<BoxTransport>() {
+            // SAFETY: `self` is `BoxTransport`. This is a no-op.
+            let this = std::mem::ManuallyDrop::new(self);
+            return unsafe { std::mem::transmute_copy(&this) };
```

Comment:
What's the difference between this and just a regular transmute?

Reply:
transmute doesn't allow size mismatch, T has generic size so it never works

Reply:
I see, so we effectively just copy the pointer here? hence the manual drop

Reply:
yeah